cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/ReplicationStatusKey.java

```diff
@@ -26,18 +26,28 @@
  */
 public final class ReplicationStatusKey {
 
+  private final String rowType;
+  private final String regionName;
+  private final String rsID;
   private final byte[] key;
 
   public ReplicationStatusKey(byte[] rowType, byte[] regionName, byte[] regionServerID) {
     byte[] rowKey;
     rowKey = Bytes.add(Bytes.toBytes(rowType.length), rowType);
     rowKey = Bytes.add(rowKey, Bytes.toBytes(regionName.length), regionName);
     rowKey = Bytes.add(rowKey, Bytes.toBytes(regionServerID.length), regionServerID);
+    this.rowType = Bytes.toString(rowKey);
+    this.regionName = Bytes.toString(regionName);
+    this.rsID = Bytes.toString(regionServerID);
     this.key = rowKey;
   }
 
   public ReplicationStatusKey(byte[] key) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(key);
     this.key = key;
+    this.rowType = getString(byteBuffer);
+    this.regionName = getString(byteBuffer);
+    this.rsID = getString(byteBuffer);
```

Comment:
please add a unit test to make sure encoding/decoding is done properly in this class.